types/react-hyperscript-helpers/react-hyperscript-helpers.errors.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-unused-vars */
+import { ReactElement } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+
+// Invalid prop
+// THROWS Argument of type '{ invalidProp: boolean; }' is not assignable to parameter of type 'ReactNode[] | WithDataAttributes<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>> | undefined'
+div({ invalidProp: true })
+
+// THROWS Argument of type '{ className: string; }' is not assignable to parameter of type 'ReactNode[]'.
+div({ className: 'class-a' }, { className: 'class-b' })
+
+// THROWS Argument of type 'string[]' is not assignable to parameter of type 'Omit<WithDataAttributes<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>>, "children">'.
+div(['Content'], ['More content'])
+
+interface TestComponentProps {
+  stringProp: string
+  optionalNumberProp?: number
+}
+
+const TestComponent = (props: TestComponentProps) => div()
+
+// Missing required prop
+// TODO: THROWS error
```

Comment:
What is the TODO for?

Reply:
Since TestComponent is defined as:

```ts
interface TestComponentProps {
  stringProp: string
  optionalNumberProp?: number
}

const TestComponent = (props: TestComponentProps) => div()
```

Ideally, TS would recognize that this is missing a required prop (`stringProp`).

```ts
h(TestComponent)
```

But currently, it accepts this as valid.

Reply:
I don't really understand how this test file works. Does the TODO prefix prevent this test case from failing?

Reply:
Yes. Negative test cases have a comment line that begins with `// THROWS expected error message`.

Reply:
Clarified in https://github.com/DataBiosphere/terra-ui/pull/3404/commits/f1a40878bf52c307216a899c47ac28187ecc7e3c.